bigquery-connector-common/src/main/java/com/google/cloud/bigquery/connector/common/BigQueryClientFactory.java

```diff
@@ -111,10 +111,12 @@ public int hashCode() {
     // ExternalAccountCredentials first and then compare their hashCodes.
     if (credentials instanceof ExternalAccountCredentials) {
       return Objects.hashCode(
-          BigQueryUtil.getCredentialsByteArray(credentials), headerProvider, bqConfig);
+          BigQueryUtil.getCredentialsByteArray(credentials),
+          headerProvider,
+          new BigQueryClientFactoryConfig(bqConfig));
```

Comment:
To avoid duplication, please move this conversion to the constructor (line 71)

Reply:
Ack